### PR TITLE
Issue #2982332 by Insasse: New Riddle API

### PR DESCRIPTION
**Pull request by ol0lll**

### CODE REVIEW

File: riddle_marketplace.post_update.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * @file
+ * Post update functions.
+ */
+
+/**
+ * Set new api url if needed.
+ */
+function riddle_marketplace_post_update_api_url() {
+  $module_settings = Drupal::configFactory()->get('riddle_marketplace.settings');
```
Comment (DiDebru) [inline riddle_marketplace.post_update.php]: We need Drupal::configFactory()->getEditable(); here

File: riddle_marketplace.post_update.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  $api_url = $module_settings->get('riddle_marketplace.api_url');
+  $new_api_url = 'https://www.riddle.com/api/creators/riddle/get-list?token=%%TOKEN%%';
+
+  if ($api_url != $new_api_url) {
+    $module_settings->set('riddle_marketplace.api_url', $new_api_url);
+  }
```
Comment (DiDebru) [inline riddle_marketplace.post_update.php]: And we need to save the settings here
$module_settings->save();

Comment (DiDebru) [reply riddle_marketplace.post_update.php]: between line 17 and 18.

**Comment by DiDebru** [comment]
Now its working fine for me thanks for the patch. I've also added a tip in the comment section to export the config after this update.

File: modules/media_riddle_marketplace/src/Plugin/MediaEntity/Type/Riddle.php [hunk 3]
```diff
@@ -163,27 +177,27 @@ public function getField(MediaInterface $media, $name) {
...
+            $directory = $this->configFactory->get('media_riddle_marketplace.settings')->get('local_images');
+            // Try to load from local filesystem.
+            $absolute_uri = current(glob($this->fileSystem->realpath($directory . '/' . $code . ".*{jpg,jpeg,png,gif}"), GLOB_BRACE));
+            if ($absolute_uri) {
+              return $directory . '/' . $this->fileSystem->basename($absolute_uri);
+            }
+            file_prepare_directory($directory, FILE_CREATE_DIRECTORY | FILE_MODIFY_PERMISSIONS);
+            // Get image from remote and save locally.
+            try {
+              $response = \Drupal::httpClient()->head($riddle['image']);
```
Comment (dbosen) [inline modules/media_riddle_marketplace/src/Plugin/MediaEntity/Type/Riddle.php]: Please inject this

File: modules/media_riddle_marketplace/src/Plugin/MediaEntity/Type/Riddle.php [hunk 4]
```diff
@@ -163,27 +177,27 @@ public function getField(MediaInterface $media, $name) {
...
+              $format = ExtensionGuesser::getInstance()
```
Comment (dbosen) [inline modules/media_riddle_marketplace/src/Plugin/MediaEntity/Type/Riddle.php]: You can use a drupal core service for that, either file.mime_type.guesser.extension or file.mime_type.guesser

**Review by dbosen — CHANGES_REQUESTED** [review]

File: modules/media_riddle_marketplace/src/Plugin/MediaEntity/Type/Riddle.php [hunk 5]
```diff
@@ -236,4 +247,26 @@ public function getDefaultName(MediaInterface $media) {
...
+   * If the mime type is unknown, returns null.
+   *
+   * This method uses the mime type as guessed by getMimeType()
+   * to guess the file extension.
+   *
+   * @param string $mime_type
+   *   The mime type to guess extension for.
+   *
+   * @return string|null
+   *   The guessed extension or null if it cannot be guessed
```
Comment (DiDebru) [inline modules/media_riddle_marketplace/src/Plugin/MediaEntity/Type/Riddle.php]: Hope this is not too pedantic but every comment should end with a full stop.

**Review by dbosen — APPROVED** [review]

**Pull request merged**